### PR TITLE
Naming csv files, updated guide links.

### LINKED ISSUE #426: Drop down menu for "User Data" block not available
Thank for your for this brilliant work!

**(Reproducible) Example**

- The documentation for the Block __User Data__ says: "__drop down__: Select dataset by name." 
- I loaded a CSV file from my local machine and from a URL.
- Both times I can find the data on the right pane in the tab "Data" by clicking on the down arrow. The name of the object is the name of the CSV file.
- But, I don not get a drop down menu in the User Data block. I have to type the name of the object to get the data loaded in the block
- I have tried with Firefox 82.0.2 and Chrome 86.0.4240.111.

**Question/Bug?** Has the drop down been removed or is this an issue on my side? I can see it in the @MayaGans video demonstration at the rstudio::conf 2020 (great lighting talk!)

**Feauture request?** It would be great if it was able to name the data object as the CSV is imported. Some CSVs from URLs have long names that make it cumbersome to type them into the User Data block Is it possible to give a name to the  the imported dataset with a name other than by renaming the original CSV?

### PR DESCRIPTION
**Pull request by DarkMetroid**
Lets users name their csv files. Fixes some links on the right panel (guide links didn't work with our localization).

### CODE REVIEW

File: libs/ui/saveDialog.jsx [hunk 1]
```diff
@@ -280,7 +280,7 @@ export class SaveAllSvgFormDialog extends React.Component{
       filename: filename,
       linkId: 'downloadAllSvg',
       title: 'Save All SVGs',
-      contentText: 'Enter the name for your zip file.'
+      contentText: 'Enter the name for your zip file.',
```
Comment (gvwilson) [inline libs/ui/saveDialog.jsx:283]: why a comma at the end of this line? isn't it the last one in the object?

File: libs/ui/saveDialog.jsx [hunk 2]
```diff
@@ -348,17 +348,25 @@ export class LoadCsvDialog extends React.Component{
     this.state = {
       open: false,
       url: '',
-
+      pageName: 'default',
+      csvName: '',
```
Comment (gvwilson) [inline libs/ui/saveDialog.jsx:352]: same question - do you usually put commas at the end of the last line in an object? (should I be doing this?)

File: libs/ui/saveDialog.jsx [hunk 3]
```diff
@@ -370,53 +378,116 @@ export class LoadCsvDialog extends React.Component{
...
   localFileUpload (fileUploadRef){
-    fileUploadRef.click()
-    this.handleClose()
+    this.refs.csvFile.click()
   }
 
   urlLoad(){
-    this.props.loadCsvUrl(this.state.url)
+    // Get the end of the url, removing the extension (if it's there)
+    const formattedUrl = this.state.url.replace(/#[^#]+$/, "").replace(/\?[^\?]+$/, "").replace(/\/$/, "");
```
Comment (gvwilson) [inline libs/ui/saveDialog.jsx:392]: whoa - isn't there a standard function somewhere that does this cleanup?

File: libs/ui/saveDialog.jsx [hunk 4]
```diff
@@ -370,53 +378,116 @@ export class LoadCsvDialog extends React.Component{
...
   render () {
     return (
       <Dialog open={this.state.open} onClose={this.handleClose} aria-labelledby="form-dialog-title">
+        <input type="file" id="csvFile" ref="csvFile"
+          onChange={this.loadCsv}
+          style={{display: "none"}}/>
+
         <DialogTitle id="form-dialog-title">Load CSV</DialogTitle>
+        { this.state.pageName == 'default' ?
+          <>
```
Comment (gvwilson) [inline libs/ui/saveDialog.jsx:438]: I don't know what this is, but that's probably my limited knowledge of React.

File: libs/ui/tabs.jsx [hunk 5]
```diff
@@ -206,7 +206,7 @@ function MultiDataGrid (props) {
         <div className="emptyPanelWrapper">
           <FontAwesomeIcon className="panelIcon"icon={faDatabase} />
           <p className="emptyPanelText">You aren't currently displaying any {props.descriptorLabel}.</p>
-          <p className="emptyPanelText"> To learn more about getting started visit&nbsp;<a href="/guide/" className="guideLink">our guide</a>.
+          <p className="emptyPanelText"> To learn more about getting started visit&nbsp;<a href="./guide/" className="guideLink">our guide</a>.
```
Comment (gvwilson) [inline libs/ui/tabs.jsx:209]: Good catch.

**Review by gvwilson — APPROVED** [review]
A couple of very minor questions - let me know and then I'll merge?

File: libs/ui/saveDialog.jsx (hunk 1 above)
Comment (DarkMetroid) [reply libs/ui/saveDialog.jsx:283]: It's valid EMCA, but I think it's really up to us what convention we prefer so long as it's consistent.

In general if each entry is on it's own line I leave the last comma per the following lint: https://eslint.org/docs/rules/comma-dangle#:~:text=Trailing%20commas%20in%20object%20literals,5%20(and%20ECMAScript%203!)&text=However%2C%20IE8%20(when%20not%20in,encounters%20trailing%20commas%20in%20JavaScript.

File: libs/ui/saveDialog.jsx (hunk 4 above)
Comment (DarkMetroid) [reply libs/ui/saveDialog.jsx:438]: Empty fragment, so with React we need one element wrapping everything there. The <> shortthand maps to <React.Fragment>. Avoid creating a DOM element we don't need, while satisfying React.

File: libs/ui/saveDialog.jsx (hunk 3 above)
Comment (DarkMetroid) [reply libs/ui/saveDialog.jsx:392]: Nothing standard unfortunately, that was the recommended approach on Stackoverflow. I think the messiness comes from having to handle stripping query parameters etc.

Interestingly an Rstudio link came up when I was googling for packages: https://community.rstudio.com/t/get-file-name-from-url/25505

Don't see anything offhand that does what we need with javascript/React unfortunately.

**Pull request merged**